web-registry/src/pages/Home/Home.loader.ts

```diff
@@ -0,0 +1,74 @@
+import { QueryClient } from '@tanstack/react-query';
+
+import { getEcocreditQueryClient } from 'lib/clients/ecocreditQueryClient';
+import { getMarketplaceQueryClient } from 'lib/clients/marketplaceQueryClient';
+import { getSimplePriceQuery } from 'lib/queries/react-query/coingecko/simplePrice/simplePriceQuery';
+import { getProjectsQuery } from 'lib/queries/react-query/ecocredit/getProjectsQuery/getProjectsQuery';
+import { getAllowedDenomQuery } from 'lib/queries/react-query/marketplace/getAllowedDenomQuery/getAllowedDenomQuery';
+import { getSellOrdersExtentedQuery } from 'lib/queries/react-query/marketplace/getSellOrdersExtentedQuery/getSellOrdersExtentedQuery';
+import { getAllCreditClassesQuery } from 'lib/queries/react-query/sanity/getAllCreditClassesQuery/getAllCreditClassesQuery';
+import { getAllHomePageQuery } from 'lib/queries/react-query/sanity/getAllHomePageQuery/getAllHomePageQuery';
+import { getFromCacheOrFetch } from 'lib/queries/react-query/utils/getFromCacheOrFetch';
+
+import { client as sanityClient } from '../../sanity';
+
+type LoaderType = {
+  queryClient: QueryClient;
+};
+
+export const homeLoader =
+  ({ queryClient }: LoaderType) =>
+  async ({ params }: { params: any }) => {
+    // Clients
+    const ecocreditClient = await getEcocreditQueryClient();
+    const marketplaceClient = await getMarketplaceQueryClient();
+
+    // Queries
+    const allHomePageQuery = getAllHomePageQuery({ sanityClient });
+    const allCreditClassesQuery = getAllCreditClassesQuery({ sanityClient });
+    const simplePriceQuery = getSimplePriceQuery({});
+    const projectsQuery = getProjectsQuery({
+      client: ecocreditClient,
+      request: {},
+    });
+    const sellOrdersQuery = getSellOrdersExtentedQuery({
+      client: marketplaceClient,
+      reactQueryClient: queryClient,
+      request: {},
+    });
+    const allowedDenomQuery = getAllowedDenomQuery({
+      client: marketplaceClient,
+    });
+
+    // Fetch or Cache
+
+    // Mandatory data
+    await getFromCacheOrFetch({
+      query: allHomePageQuery,
+      reactQueryClient: queryClient,
```

Comment:
just a thought i wanted to share, we have this same line quite a few times, and i was thnking that if we wanted to reduce that duplication, we could refactor this into class, where we move the `queryClient` into a constructor. but i don't feel very strongly about implementing that right now

Reply:
That would help with the loader case indeed where there's a lot of usages maybe a bit less for a single call `(new getFromCacheOrFetch(queryClient)).getQuery(allHomePageQuery)`.
I'll give it some thought, we can always improve the loaders later on with something like that and keep the other call elsewhere.